src/PhpSpec/Locator/PSR0/PSR0Locator.php

```diff
@@ -357,7 +357,7 @@ private function createResourceFromSpecFile($path)
      */
     private function validatePsr0Classname($classname)
     {
-        $pattern = '/^([a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*[\/\\\\]?)*[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*$/';
+        $pattern = '/^[^\r\n]([a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*[\/\\\\]?)*[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*$/';
```

Comment:
Ah ok it might be that `preg_match` is not in multiline mode so `^` and `$` are matching 'start of line' and 'end of line', that means extra linebreaks are allowed before and after.

You could try using `\A` and `\z` instead (note not `\Z` as that would allow a linebreak at the end), or try the `/m` or `/D` switches at the end of the pattern to affect the default behaviour of `^` and `$`.